src/Foundatio/Queues/SharedQueueOptions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Foundatio.Serializer;
+using Foundatio.Utility;
+using Microsoft.Extensions.Logging;
+
+namespace Foundatio.Queues {
+    public interface ISharedQueueOptions {
+        string Name { get; set; }
+        int Retries { get; set; }
+        TimeSpan WorkItemTimeout { get; set; }
+    }
+
+    public class SharedQueueOptions<T> : SharedOptions, ISharedQueueOptions where T : class {
+        public string Name { get; set; } = typeof(T).Name;
+        public int Retries { get; set; } = 2;
+        public TimeSpan WorkItemTimeout { get; set; } = TimeSpan.FromMinutes(5);
+        public IEnumerable<IQueueBehavior<T>> Behaviors { get; set; }
+    }
+
+    public interface ISharedQueueOptionsBuilder : ISharedOptionsBuilder {}
+
+    public static class SharedQueueOptionsBuilderExtensions {
```

Comment:
There should be extension methods for queue behaviors.

Reply:
The generic argument for behaviors is the critical block for the builder design pattern

Reply:
Yeah, the behavior methods were causing all kinds of problems. But I wonder if it will be better if the methods are on the shared options class directly instead of being extension methods? I will give it a try.

Reply:
I added the behavior methods to the concrete builder class. It will need to be added to each implementation.